### DISCUSSION TITLE
Using ScratchAddons on ScratchLab

### DISCUSSION THREAD

**Discussion by sfederici** [ScratchAddons/ScratchAddons#5197] [Discussion]
Would it be very difficult to run ScratchAddons on the accessible Scratch version available at Scratch Lab (https://lab.scratch.mit.edu/contrast-blocks/)? If feasible, what should I change in the ScratchAddons code in order to manually install it on this website?

Thanks in advance for any help

**Comment by WorldLanguages** [comment]
It's possible by changing some parts of the source.
But wouldn't it be better if a combination of addons allowed you to have the contrast blocks in actual Scratch?

**Reply by WorldLanguages** [reply]
Btw, offtopic - you're a teacher, right? It would be great if you could summarize what you find best about using Scratch Addons with your students. The average SA user installs SA by themselves, not because the teacher required it, so we often don't take teachers too much into consideration.

**Reply by sfederici** [reply]
> wouldn't it be better if a combination of addons allowed you to have the contrast blocks in actual Scratch?

Of course. An accessible color combination in the block color addon would be great. But I need it to make an experiment in these days, so I was looking for something "quick and dirt".

**Reply by sfederici** [reply]
> you're a teacher, right? It would be great if you could summarize what you find best about using Scratch Addons with your students. The average SA user installs SA by themselves, not because the teacher required it, so we often don't take teachers too much into consideration.

This is something that I'm doing these days for addressing the needs of people with learning disabilities (that is why I'm testing the accessible version in Scratch Lab). So, sure. I would gladly do that. My students are indeed using a specific set of addons (that is why I was the one that asked for the ScratchAddons' settings import/export feature) that I find particularly useful for them

**Reply by sfederici** [reply]
> It's possible by changing some parts of the source.

Would that be limited to a small part of the code and quick to do?

**Reply by apple502j** [reply]
Not that quick, as Scratch Addons expects that the website is hosted at scratch.mit.edu. But not that impossible, maybe.
One very big difference is that Scratch Lab page has the landing page and the editor in the same page, which could break some code that expects only the editor to be in the editor page.

**Reply by Secret-chest** [reply]
> > wouldn't it be better if a combination of addons allowed you to have the contrast blocks in actual Scratch?
> 
> Of course. An accessible color combination in the block color addon would be great. But I need it to make an experiment in these days, so I was looking for something "quick and dirt".

It isn't even accesible for colourblind users, because the colours are even more similar now.

**Reply by sfederici** [reply]
> Not that quick, as Scratch Addons expects that the website is hosted at scratch.mit.edu. But not that impossible, maybe. One very big difference is that Scratch Lab page has the landing page and the editor in the same page, which could break some code that expects only the editor to be in the editor page.

So, maybe it wouldn't be so easy (and fast)

**Reply by sfederici** [reply]
> It isn't even accesible for colourblind users, because the colours are even more similar now.

I guess this can be true, but the accessibility, at least in this version, is towards Dyslexic people. It has been proved that they can read better when the background is a pastel color

**Reply by sfederici** [reply]
> Btw, offtopic - you're a teacher, right? It would be great if you could summarize what you find best about using Scratch Addons with your students

This is the list of addons I enabled in the settings that my students import, so that everyone can see and use these features during my lessons (sorry if I'm backstranslating the addon names, my browser is in Italian):

- search bar in the editor
- disable automatic saving (to avoid that "experiments" are permanently saved)
- automatically add extensions (we use the -once standard- music and pen blocks, so I don't have to go into extensions)
- progress bar for projects (so that they don't get anxious)
- confirm sprite removal (so that they aren't removing things by mistake)
- fix SVG loading (so they don't complain about "wrong" SVG)
- debugger (so I can show them what happens when running a script step-by-step and how do scripts interact with their yield rules)
- editor mode with disabled audio (so I can show them projects with audio and temporarely mute them when I have to speak)
- hide stage and sprite area (so they have a larger coding area even with small screens)
- automatically hide block palette (so they have a larger coding area even with small screens)
- do not automatically run duplicated scripts (so they don't get confused)
- onion skin (so creating costumes for animations is easier)
- mouse position (so they can easily learn and experiment with stage coordinates)
- pause button (so I can temporarely stop a project when I'm explaining how it works)
- block replace (useful, but I would like it to be more extended, so that all "similar" blocks can be replaced)
- move costumes to the top/bottom (to avoid to drag costumes a lot when there are complex animations based on costumes)
- developer tools (to paste the script at the mouse position)
- dragging file (useful as my students like a lot dragging, instead of importing)
- block opacity (so they can clearly see where they are going to embed a reporter)

As you see, with my students I use editor addons only. They are free to experiment with other addons, but they are asked not to change these settings as we use them during lessons.

**Reply by WorldLanguages** [reply]
Thanks!!

> (sorry if I'm backstranslating the addon names, my browser is in Italian):

See #5202 :D

**Reply by WorldLanguages** [reply]
@sfederici We read your [Is the Scratch Programming Environment Ideal for all?](https://repositorio.grial.eu/bitstream/grial/2886/1/scratch.pdf) publication. Very interesting!

**Reply by WorldLanguages** [reply]
Any tips you'd give to other teachers using Scratch Addons in their classroom (other than the enabled features themselves)? Of course I can imagine the import/export settings feature is useful. Anything else?

**Reply by sfederici** [reply]
> @sfederici We read your [Is the Scratch Programming Environment Ideal for all?](https://repositorio.grial.eu/bitstream/grial/2886/1/scratch.pdf) publication. Very interesting!

I'm glad you liked it. It was inspired by this very discussion

**Reply by sfederici** [reply]
> Any tips you'd give to other teachers using Scratch Addons in their classroom (other than the enabled features themselves)? Of course I can imagine the import/export settings feature is useful. Anything else?

Yes, you are right, the import/export settings feature (that I asked to add to ScratchAddons but that it is not an improvement to Scratch) is it the very base to the possibility to use ScratchAddons in a classroom. 

> Anything else?

I'll think about it

**Comment by Cool-Scratcher** [comment]
I don't understand why this would even be wanted. Who's using Scratch Lab often enough that they want SA enabled on it...

**Reply by BroJac5246** [reply]
> I don't understand why this would even be wanted. Who's using Scratch Lab often enough that they want SA enabled on it...

But cumulatively...

**Reply by sfederici** [reply]
> I don't understand why this would even be wanted. Who's using Scratch Lab often enough that they want SA enabled on it...

As you can see, this is not a request for an addon. This is just a "quick and dirty" (if possibile) hack to ScratchAddons to use it in a current experiment. But I guess that hacking the "block color" addon and installing ScratchAddons manually would be easier and faster.

**Comment by sfederici** [comment]
> See #5202 :D
This is something I really needed :)Thanks for letting me know
